test/function_wrappers.ts

```diff
@@ -77,20 +77,23 @@ describe('wrapUserFunction', () => {
   it('correctly wraps an async CloudEvent function', done => {
     const request = createRequest(CLOUD_EVENT);
     const response = createResponse();
-    const func = wrapUserFunction(async (cloudEvent: CloudEvent) => {
-      assert.deepStrictEqual(cloudEvent, CLOUD_EVENT);
-      // await to make sure wrapper handles async code
-      await new Promise(resolve => setTimeout(resolve, 20));
-      done();
-    }, 'cloudevent');
+    const func = wrapUserFunction(
+      async (cloudEvent: CloudEvent<Record<string, string>>) => {
+        assert.deepStrictEqual(cloudEvent, CLOUD_EVENT);
+        // await to make sure wrapper handles async code
+        await new Promise(resolve => setTimeout(resolve, 20));
+        done();
+      },
+      'cloudevent'
+    );
     func(request, response, () => {});
   });
 
```

Comment:
I think one of the great benefits is using this type with a more complicated interface.

Can we add a test for `CloudEvent<MyData>` where `MyData` is an interface we provide? This is probably how developers would this (only TS right?)

Reply:
Good idea. I will add an integration test that covers a more complex type hint.

Reply:
Added a test in test/integration/cloud_event.ts